endpoints/api_config.py

```diff
@@ -1914,23 +1918,29 @@ def __api_descriptor(self, services, hostname=None):
 
     return descriptor
 
-  def get_descriptor_defaults(self, api_info, hostname=None):
+  def get_descriptor_defaults(self, api_info, hostname=None, local=True):
     """Gets a default configuration for a service.
 
     Args:
       api_info: _ApiInfo object for this service.
       hostname: string, Hostname of the API, to override the value set on the
         current service. Defaults to None.
+      local: bool, True if locally generated. If true, protocol will be
+        determined by whether the host is localhost. Otherwise, it will be
+        determined by whether the app is running on a dev server.
 
     Returns:
       A dictionary with the default configuration.
     """
     hostname = (hostname or endpoints_util.get_app_hostname() or
                 api_info.hostname)
-    protocol = ('http' if hostname and hostname.startswith('localhost') else
-                'https')
-    protocol = ('http' if hostname and hostname.startswith('localhost') else
-                'https')
+    if local:
+      protocol = ('http' if hostname and hostname.startswith('localhost') else
+                  'https')
+    else:
+      protocol = ('http' if endpoints_util.is_running_on_devserver() else
```

Comment:
As I understand it, a config can be generated either remotely (via an HTTP call to the service) or locally (using endpointscfg). If it's done locally, then it doesn't make sense to check whether we're running on a dev_appserver or not; it's just being run from a command-line prompt.

Does that make sense?